pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2355,6 +2374,18 @@ type NodeStatus struct {
 
 	// SynchronizationStatus is the synchronization status of the node
 	SynchronizationStatus *NodeSynchronizationStatus `json:"synchronizationStatus,omitempty" protobuf:"bytes,25,opt,name=synchronizationStatus"`
+
+	// TaskResultSynced is used to determine if the node's output has been received
+	TaskResultSynced *bool `json:"taskResultSynced,omitempty" protobuf:"bytes,28,opt,name=taskResultSynced"`
```

Comment:
Why make this a pointer to a boolean rather than just a boolean? Reading the PR, I don't see where having the extra state of the `nil` condition gets you that much to where it justifies the extra complexity of doing the nil checks and setting the references